File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.cpp
class Solution {
public:
  long long countBadPairs(vector<int>& nums) {
    long long ans = 0;
    unordered_map<int, int> m;
    for (int i = 0; i < nums.size(); i++) {
        ans += i - m[nums[i] - i];
        m[nums[i] - i]++;
    }
    return ans;
}
};

File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.java
class Solution {
    public long countBadPairs(int[] nums) {
         long pair = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int diff = nums[i] - i;
            pair += i - map.getOrDefault(diff, 0);
            map.put(diff, map.getOrDefault(diff, 0) + 1);
        }
        return pair;
    }
}